common/OpTestSSH.py

```diff
@@ -155,52 +159,133 @@ def get_console(self):
+        console.expect(expect_prompt)
+        combo_io = (console.before + console.after).lstrip()
+        set_env_list += combo_io.splitlines()
+        # remove the expect prompt since matched generic #
+        del set_env_list[-1]
+        return set_env_list
+
+    def try_sendcontrol(self, console, command):
+        console.sendcontrol('c')
+        try_list = []
+        try_list += self.set_env(console)
+        expect_prompt = self.build_prompt() + "$"
+        # we need the env to get the expect_prompt
+        rc = console.expect([expect_prompt, pexpect.TIMEOUT, pexpect.EOF], 10)
+        if rc != 0:
+          self.terminate()
+          raise CommandFailed(command, 'run_command TIMEOUT', -1)
         else:
-            prompt = "\[console-pexpect\]#"
-        expect_prompt = prompt + "$"
+          try_list += console.before.splitlines()
+          echo_rc = 1
+        return try_list, echo_rc
 
-        console.sendline(command)
-        console.expect("\n") # from us
-        rc = None
-        output = None
-        exitcode = None
-        try:
-            rc = console.expect([expect_prompt], timeout=timeout)
-            output = console.before
-            console.sendline("echo $?")
-            console.expect("\n") # from us
-            rc = console.expect([expect_prompt], timeout=timeout)
-            exitcode = int(console.before)
-        except pexpect.TIMEOUT as e:
-            print e
-            print "# TIMEOUT waiting for command to finish."
-            print "# Attempting to control-c"
-            try:
-                console.sendcontrol('c')
-                rc = console.expect(["\[console-pexpect\]#$"], 10)
-                if rc == 0:
-                    raise CommandFailed(command, "TIMEOUT", -1)
-            except pexpect.TIMEOUT:
-                print "# Timeout trying to kill timed-out command."
-                print "# Failing current command and attempting to continue"
-                self.terminate()
-                raise CommandFailed("ssh -p %u" % self.port, "timeout", -1)
-            raise e
+    def retry_password(self, console, command):
+        retry_list_output = []
+        a = 0
+        while a < 3:
+          a += 1
+          console.sendline(self.password)
+          rc = console.expect([".*#", "try again.", pexpect.TIMEOUT, pexpect.EOF])
+          if (rc == 0) or (rc == 1):
+            combo_io = console.before + console.after
+            retry_list_output += combo_io.splitlines()
+            matching = [xs for xs in sudo_responses if any(xs in xa for xa in console.after.splitlines())]
+            if len(matching):
+              echo_rc = 1
+              rc = -1 # use to flag the failure next
+          if rc == 0:
+            retry_list_output += self.set_env(console)
+            echo_rc = 0
+            break
+          elif a == 2:
+            echo_rc = 1
+            break
+          elif (rc == 2) or (rc == 3):
```

Comment:
Here also do the same handling for EOF raise "Session EOF" exception and set the state to disconnected.

Reply:
Fixed